HW1/P3/P3.py

```diff
@@ -0,0 +1,19 @@
+#NOTA BENE: this code has to be copied and pasted line by line in the PySpark console of AWS, this is how I got mu result. It won't run with the commant python P3.py
+
+#loading data from the cluster
+wlist = sc.textFIle('s3://Harvard-CS205/wordlist/EOWL_words.txt')
+
+#duplicating to form key value pairs
+w_w = w.map(lambda w:(w,w))
+
+#sorting the key
+s_w = w_w.map(lambda (k,v): (''.join(sorted(k)),v))
```

Comment:
I think for here: sorted() in python takes O(nlogn) time complexity. If we have a lot of data, that would be not that efficiency. You can use counting sort which is linear time sorting. But for this question, it is fine to use any sorting algorithm.